database/migrations/2023_12_19_081112_fix_language_dirs.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use \App\Helpers\Helper;
+use \App\Models\Setting;
+use \App\Models\User;
+
+class FixLanguageDirs extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        /**
+         * Update the settings table
+         */
+        $settings = Setting::getSettings();
+        if (($settings) && ($settings->locale != '')) {
+            DB::table('settings')->update(['locale' => Helper::mapLegacyLocale($settings->locale)]);
```

Comment:
Is it possible that someone could have a locale set that isn't handled by the `mapLegacyLocale` method and this line results in NULL being entered into the field? I think they would have had to manually mangle their data to get there but I think it is something to consider.

Reply:
> Is it possible that someone could have a locale set that isn't handled by the mapLegacyLocale method 

They would have to have added it themselves directly in the database, in which case they can do it again (IMHO)